### PR TITLE
luarocks: make the torch binding as a torch package.

### PR DESCRIPTION
**Pull request by huxuan**
Also with minor change in `test.lua` and `util.lua`.

### CODE REVIEW

**Comment by huxuan** [comment]
@feiga @chivee Please help me to confirm the information in `multiverso-scm-1.rockspec` is correct.

**Comment by huxuan** [comment]
The installation feature also has been tested by Xiang Li. He is using the torch binding to run the experiments now. I will follow up in case of any problem. Please feel free to point out if you have any concerns with the modification.

**Comment by you-n-g** [comment]
I think it is OK

**Comment by you-n-g** [comment]
@feiga @chivee   If the infomation in `multiverso-scm-1.rockspec` is correct, I think we could merge it.

File: binding/lua/multiverso-scm-1.rockspec [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+source = {
+    url = "https://github.com/Microsoft/multiverso"
+}
+
+description = {
+    summary = "Torch binding for multiverso.",
+    detailed = [[
+        Multiverso is a parameter server framework for distributed machine
+        learning, this package make it possible to have multiverso embedded in
+        your original torch programs.
```
Comment (chivee) [inline binding/lua/multiverso-scm-1.rockspec]: Multiverso is a parameter server framework for distributed machine learning. this package can leverage multiple machines and GPU to speed-up the torch programs.

File: binding/lua/multiverso-scm-1.rockspec [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    ]],
+    homepage = "http://www.dmtk.io",
+    license = "MIT"
+}
+
+dependencies = {
+    "torch >= 7.0"
+}
+
+build = {
```
Comment (chivee) [inline binding/lua/multiverso-scm-1.rockspec]: does the build will executed under root folder as default?

**Comment by chivee** [comment]
please also offer some examples to show our ability and usage.

File: binding/lua/multiverso-scm-1.rockspec (hunk 1 above)
Comment (huxuan) [inline binding/lua/multiverso-scm-1.rockspec]: Done.

File: binding/lua/multiverso-scm-1.rockspec (hunk 2 above)
Comment (huxuan) [inline binding/lua/multiverso-scm-1.rockspec]: Currently, the build is supposed to run as the folder where it exists. Same behaviour as having it as a standalone torch package.

**Comment by huxuan** [comment]
> please also offer some examples to show our ability and usage.

I have add examples in other pull requests, and another one (the resnet binding) is already on the way.

**Comment by chivee** [comment]
things looks fine for me

**Pull request merged**